### ISSUE TITLE
Feature request: support multiple url in case websites are down

### ISSUE THREAD

**Issue by davydden** [spack/spack#11030]
extend `url` to be a list of urls as opposed to a single one.

### Rationale

oftentimes fetching fails due to websites being down:
```
==> Fetching from https://ftpmirror.gnu.org/texinfo/texinfo-6.5.tar.gz failed.
==> Error: FetchError: All fetchers failed for texinfo-6.5-jfsqfa7c3igtjeirfpf3oavvmkw5v5ir
```
a simple hack is to edit the package to use an alternative source:
```
url      = "https://ftp.gnu.org/gnu/texinfo/texinfo-6.5.tar.gz"
```
A new user might not know what to do and what to edit, so it would be good to minimize such hacks.

### Description

Not sure how intrusive/diffucult is it, but Spack could support things like
```
url    = ["https://ftpmirror.gnu.org/texinfo/texinfo-6.5.tar.gz", "https://ftp.gnu.org/gnu/texinfo/texinfo-6.5.tar.gz", ....]
```

and try to use the another source in the list if fetching failed for the current one.

### Additional information

Same issue has been raised in [Google forum](https://groups.google.com/forum/?fromgroups#!topic/spack/MW67IXrVA2w) and also in Slack recently.

**Comment by mwkrentel** [comment]
In this case, `ftpmirror.gnu.org` is itself a revolving URL location,
right?  So, I'm kinda surprised that it would fail at all.

For URLs with builtin mirrors like this, maybe all spack has to do is
retry the same URL 3-4 times before giving up.

Anyway, I very much like the idea of backup URLs where possible.

**Comment by citibeth** [comment]
This problem comes up from time to time.  As the number of packages you rely upon for your build increases, the chance that you can download them all, from their primary source, on any given day, goes to zero.

**Comment by alalazo** [comment]
Fixed by #13881
